pubsub.go

```diff
@@ -162,6 +167,41 @@ func (ps *PubSub) Publish(m Message) error {
 func (ps *PubSub) Receive(subscription string, messages chan<- *pubsub.Message) error {
 	ctx := context.Background()
 	s := ps.client.Subscription(subscription)
+	s.ReceiveSettings = pubsub.DefaultReceiveSettings
+
+	return s.Receive(ctx, func(ctx context.Context, m *pubsub.Message) {
+		messages <- m
+	})
+}
+
+func (ps *PubSub) ReceiveWithSettings(subscription string, rs ReceiveSettings, messages chan<- *pubsub.Message) error {
+	ctx := context.Background()
+	s := ps.client.Subscription(subscription)
+
+	s.ReceiveSettings = pubsub.DefaultReceiveSettings
+
+	// override the default settings with the provided settings
+	if rs.Settings.MaxExtension != 0 {
+		s.ReceiveSettings.MaxExtension = rs.Settings.MaxExtension
+	}
+	if rs.Settings.MaxExtensionPeriod != 0 {
+		s.ReceiveSettings.MaxExtensionPeriod = rs.Settings.MaxExtensionPeriod
+	}
+	if rs.Settings.MinExtensionPeriod != 0 {
+		s.ReceiveSettings.MinExtensionPeriod = rs.Settings.MinExtensionPeriod
+	}
+	if rs.Settings.MaxOutstandingMessages != 0 {
+		s.ReceiveSettings.MaxOutstandingMessages = rs.Settings.MaxOutstandingMessages
+	}
+	if rs.Settings.MaxOutstandingBytes != 0 {
+		s.ReceiveSettings.MaxOutstandingBytes = rs.Settings.MaxOutstandingBytes
+	}
+	if rs.Settings.NumGoroutines != 0 {
+		s.ReceiveSettings.NumGoroutines = rs.Settings.NumGoroutines
+	}
+
+	// always false to allow the library to use streamingpull-api
+	s.ReceiveSettings.Synchronous = false
```

Comment:
I'm curious, should we instead move this block that parses and sets `ReceiveSettings` to the `NewPubSub` client creation method, rather than having a separate function? I think a lot of our projects are already attempting to set `ReceiveSettings` when calling the `NewPubSub` function, even if those settings don't seem to be used.

Reply:
that's an interesting approach... let me play around and test it out! that you for for the feedback!